consensus/src/chained_bft/network.rs

```diff
@@ -482,4 +507,18 @@ impl<T: Payload> NetworkTask<T> {
             .send(Ok(response_data))
             .map_err(|_| format_err!("handling inbound rpc call timed out"))
     }
+
+    async fn process_epoch_change(
+        &mut self,
+        peer_id: AccountAddress,
+        ledger_info: LedgerInfoWithSignaturesProto,
+    ) -> failure::Result<()> {
+        let ledger_info = LedgerInfoWithSignatures::try_from(ledger_info)?;
```

Comment:
Don't we want to make sure we've executed the blockchain up to the change in question (i.e. calculate the next validator set ourselves). Also curious how this logic will overlap with state synchronization (which will also need to do a similar processing)

Reply:
Yeah, that's in coming PRs. Basically after it passed the networking layer, event_processor will make sure it syncs to the version of the ledger info before start new epoch.

Here in network we only handles same epoch messages while for future epoch messages, we'll probably 1. state sync to the epoch 2. pass it down here 3. main loop starts new epoch. But not there yet.